### DISCUSSION TITLE
instead of one big settings file: create smaller ones inside a folder

### DISCUSSION THREAD

**Discussion by michabbb** [ryangjchandler/laravel-json-settings#8] [Ideas]
after my first test of your package, I had to realize that this `settings.json` file will become very large and if multiple processes (worker) will have to deal with reading and writing into the same file, this can cause strange problems. so I would suggest to extend your package to make it work with multiple files. 

instead of: 

`$settings->set('key', 'value');`

it would be nice to have:

`$settings->set('namespace','key', 'value');`

and `namespace` would be the name of the settings file 😏

**Comment by ryangjchandler** [comment]
The size of the file doesn't really matter does it? The amount of disk space used will almost be the same for a single file or multiple smaller files.

**Comment by ryangjchandler** [comment]
As for namespacing - this isn't something that I'll personally need so it's unlikely that I'll add it anytime soon. If you want to open a PR, I'd be more than happy to take a look.

**Reply by michabbb** [reply]
OK
